freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/FreeIpaClientFactory.java

```diff
@@ -71,25 +72,43 @@ private String toClusterProxyBasepath(String freeIpaClusterCrn) {
 
     public FreeIpaClient getFreeIpaClientForStack(Stack stack) throws FreeIpaClientException {
         LOGGER.debug("Creating FreeIpaClient for stack {}", stack.getResourceCrn());
-
-        try {
-            if (clusterProxyService.isCreateConfigForClusterProxy(stack)) {
-                return getFreeIpaClientBuilderForClusterProxy(stack).build();
-            } else {
-                return getFreeIpaClientBuilder(stack).build();
+        Status stackStatus = stack.getStackStatus().getStatus();
+        if (!Status.FREEIPA_UNREACHABLE_STATUSES.contains(stackStatus)) {
```

Comment:
please move this condition into the status as a method, so you don't have the implementation twice